src/Microsoft.OData.Core/UriParser/SemanticAst/InNode.cs

```diff
@@ -0,0 +1,117 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="InNode.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.UriParser
+{
+    #region Namespaces
+
+    using System;
+    using Microsoft.OData;
+    using Microsoft.OData.Edm;
+    using Microsoft.OData.Metadata;
+    using ODataErrorStrings = Microsoft.OData.Strings;
+
+    #endregion Namespaces
+
+    /// <summary>
+    /// Query node representing an In operator.
+    /// </summary>
+    public sealed class InNode : SingleValueNode
+    {
+        /// <summary>
+        /// Bool reference, as In always returns boolean.
+        /// </summary>
+        private readonly IEdmTypeReference boolTypeReference = EdmLibraryExtensions.GetPrimitiveTypeReference(typeof(bool));
+
+        /// <summary>
+        /// The left operand.
+        /// </summary>
+        private readonly SingleValueNode left;
+
+        /// <summary>
+        /// The right operand.
+        /// </summary>
+        private readonly CollectionNode right;
+
+        /// <summary>
+        /// Create a InNode
+        /// </summary>
+        /// <param name="left">The left operand.</param>
+        /// <param name="right">The right operand.</param>
+        /// <exception cref="System.ArgumentNullException">Throws if the left or right inputs are null.</exception>
+        /// <exception cref="ODataException">Throws if the right operand single item type isn't the same type as the left operand.</exception>
+        public InNode(SingleValueNode left, CollectionNode right)
+        {
+            ExceptionUtils.CheckArgumentNotNull(left, "left");
+            ExceptionUtils.CheckArgumentNotNull(right, "right");
+            this.left = left;
+            this.right = right;
+
+            if (this.left.GetEdmTypeReference().Definition != this.right.ItemType.Definition)

```

Comment:
>this.left.GetEdmTypeReference().Definition != this.right.ItemType.Definition [](start = 16, length = 76)

We should support the type of the right node collection being comparable to the type of the left node (including being derived from the type of the left node).  See EdmLibraryExtensions.IsAssignableFrom.

Reply:
Thanks for the catch, I think I had forgotten to jot down the point regarding derived types during code review. Regarding supporting comparable types, is there a function or API that you're specifically referring to or is what I currently have in the code of comparing the definitions sufficient?